pkg/services/serviceaccounts/manager/service.go

```diff
@@ -77,6 +112,12 @@ func (sa *ServiceAccountsService) Run(ctx context.Context) error {
 			if _, err := sa.getUsageMetrics(ctx); err != nil {
 				sa.backgroundLog.Warn("Failed to get usage metrics", "error", err.Error())
 			}
+		case <-tokenCheckTicker.C:
+			sa.backgroundLog.Debug("checking for leaked tokens")
+
+			if err := sa.secretScanService.CheckTokens(ctx); err != nil {
+				sa.backgroundLog.Warn("Failed to check for leaked tokens", "error", err.Error())
+			}
```

Comment:
let's say we would want to initiate service accounts service without the secretScanService.

Is it possible to refactor so that we could instantiate the chan only then.